addons/binding/org.openhab.binding.innogysmarthome/src/main/java/org/openhab/binding/innogysmarthome/internal/InnogyWebSocket.java

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.innogysmarthome.internal;
+
+import java.net.URI;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.eclipse.jetty.websocket.api.Session;
+import org.eclipse.jetty.websocket.api.StatusCode;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketClose;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketConnect;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketError;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketMessage;
+import org.eclipse.jetty.websocket.api.annotations.WebSocket;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+import org.openhab.binding.innogysmarthome.handler.InnogyBridgeHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link InnogyWebSocket} implements the websocket for receiving constant updates
+ * from the innogy SmartHome web service.
+ *
+ * @author Oliver Kuhl - Initial contribution
+ */
+@WebSocket
+public class InnogyWebSocket {
+
+    private Logger logger = LoggerFactory.getLogger(InnogyWebSocket.class);
+    private final CountDownLatch closeLatch;
+    private Session session;
+    private InnogyBridgeHandler bridgeHandler;
```

Comment:
Instead of `InnogyBridgeHandler` you may take only the interface `EventListener` here.